spacy/ml/models/entity_linker.py

```diff
@@ -117,34 +117,20 @@ def empty_kb_factory(vocab: Vocab):
 
 
 @registry.misc("spacy.CandidateGenerator.v1")
-def create_candidates() -> Callable[[KnowledgeBase, Span], Iterable[Candidate]]:
-    return get_candidates
-
-
-@registry.misc("spacy.CandidateBatchGenerator.v1")
```

Comment:
Instead of removing these, we need to keep them and add a new `"spacy.CandidateGenerator.v2"`

Reply:
I get that, but we can't just go in and change the signature and implementation of an existing v1 function - that would become impossible to document for different users using different versions of spaCy.

Reply:
Done in https://github.com/explosion/spaCy/pull/12367/commits/2951c19472d4bf17aa6821d7dfb2d194b78d081b.